ewatercycle/parameter_sets/__init__.py

```diff
@@ -0,0 +1,105 @@
+from itertools import chain
+from logging import getLogger
+from os import linesep
+from typing import Iterable
+
+from ewatercycle import CFG
+from . import _pcrglobwb, _lisflood, _wflow
+from .default import ParameterSet
+from ._example import ExampleParameterSet
+from ..config import SYSTEM_CONFIG, USER_HOME_CONFIG
+
+logger = getLogger(__name__)
+
+
+def _parse_parametersets():
+    parametersets = {}
+    for name, options in CFG["parameter_sets"].items():
+        parameterset = ParameterSet(name=name, **options)
+        parametersets[name] = parameterset
+
+    return parametersets
+
+
+def available_parameter_sets(target_model: str = None) -> Iterable[str]:
+    """List available parameter sets on this machine.
+
+    Args:
+        target_model: Filter parameter sets on a model name
+
+    Returns: Names of available parameter sets on current machine.
+
+    """
+    all_parameter_sets = _parse_parametersets()
```

Comment:
Can we print a nice message when there is no available parameterset? Currently it returns an empty generator...

Reply:
Added, still needs docs explaining how to add a pararmeter set